articles/how-to-upload-images-to-cloudinary-using-node.js/index.md

```diff
@@ -0,0 +1,337 @@
+|Vue.js|Django|iOS|
+|JavaScript|Go| -  |
+|jQuery|Java| -  |
+| -  |Node.js| -  |
+| -  |.Net| -  |
+
+You can find a comprehensive guide or tutorial on how to upload images to Cloudinary using _Django_ [here](https://www.section.io/engineering-education/uploading-images-to-cloudinary-from-django-application/#conclusion). I highly recommend it for _Python_ developers.
+
+In this tutorial, we are going to learn how to upload an image to Cloudinary using Node.js.
+
+### Key takeaways
+
+At the end of the tutorial, you will have learned about:
+
+- Getting started with Cloudinary.js
+- Integrating Cloudinary.js in a Node application
+- Uploading an image to Cloudinary
+
+### Pre-requisites
+
+The basic requirements for this tutorial are:
+
+- Web development basics.
+- A basic IDE installation on the machine. In our case, we shall use Visual Studio Code (free, easy to use, and very efficient).
+- Node basics.
+- A stable internet connection.
+
+### Table of Contents
+
+- [Create account](#create-account)
+- [Setting up our project](#setting-up-our-project)
+- [Folder structure](#folder-structure)
+- [Configure package.json](#configure-packagejson)
+- [Setting up our starting point](#setting-up-our-starting-point)
+- [Configure .env file](#configure-env-file)
+- [Run the code](#run-the-code)
+- [Conclusion](#conclusion)
+- [Further Projects](#further-projects)
+- [Further study](#further-study)
+- [References](#references)
+
+### Create account
+
+Cloudinary provides a developer with a free account option. Some of the advantages of this account are mentioned in the blog found [here](https://www.section.io/engineering-education/uploading-images-to-cloudinary-from-django-application/#conclusion).
+Once created proceed to the next steps.
+
+### Setting up our project
+
+Create a new directory named "Cloudinary-uploader". In it, create two new files, "index.js" and ".env" respectively. Open your folder and view it in your IDE. Open an integrated terminal and run within the folder `npm init -y` to quickly create the "package.json" file.
+Install the needed libraries for our project. These include:
+
+- nodemon
+- cloudinary
+- formidable
+- dotenv
+- express
+
+> [formidable.js](https://www.npmjs.com/package/formidable) library was made to focus on video and image uploading and encoding. It is used for parsing form data, mostly file uploads.
+
+To install all at once, run:
+
+```node.js
+npm i nodemon cloudinary dotenv formidable express
+```
+
+`
+npm i nodemon cloudinary dotenv formidable express
+`
+
+> [dotenv.js](https://www.npmjs.com/package/dotenv) library will enable us to easily and quickly load environment variables from a locally stored .env file into `process.env`. This provides safety when working with API names and keys in any Node.js project since they are stored away from the main process and are only available when required. It provides safety and security using [The Twelve-Factor App](https://12factor.net/config) methodology.

```

Comment:
It would be great if you could include an article that explains in detail about how env variables are used. 
https://www.section.io/engineering-education/nodejs-environment-variables/

Reply:
Done